src/Microsoft.FeatureManagement/FeatureManagerSnapshot.cs

```diff
@@ -1,23 +1,26 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
+using Microsoft.FeatureManagement.FeatureFilters;
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.FeatureManagement
 {
     /// <summary>
     /// Provides a snapshot of feature state to ensure consistency across a given request.
     /// </summary>
-    class FeatureManagerSnapshot : IFeatureManagerSnapshot
+    class FeatureManagerSnapshot : IFeatureManagerSnapshot, IVariantFeatureManagerSnapshot
     {
-        private readonly IFeatureManager _featureManager;
+        private readonly FeatureManager _featureManager;
         private readonly ConcurrentDictionary<string, Task<bool>> _flagCache = new ConcurrentDictionary<string, Task<bool>>();
+        private readonly IDictionary<string, Variant> _variantCache = new Dictionary<string, Variant>();
```

Comment:
Why is this one not concurrent dictionary like flag cache?

Reply:
I think I left this in on accident when starting from the dynamic features code, I can change to concurrent.